code-challenges.test.js

```diff
@@ -21,36 +20,146 @@
 // Example input: 10
 // Expected output: [1, 1, 2, 3, 5, 8, 13, 21, 34, 55]
 
+// 1a) Pseudo code:
+// Copy and paste test code template into assignment (make sure nothing is inside initial template)
+// Read problem statement and understand inputs/outputs (inputs need to be directly inserted into test code, and not as variables)
+// Create relevant info for describe, it, expect, and toEqual methods
+// Test code output to ensure it is failing red (should get a reference error saying function is not defined)
+
+describe("fibonator", () => {
+  it("A function which takes in a number input and returns an array of length equal to the input, filled with numbers which follow the Fibonnaci sequence", () => {
+    expect(fibonator(6)).toEqual([1, 1, 2, 3, 5, 8]);
+    expect(fibonator(10)).toEqual([1, 1, 2, 3, 5, 8, 13, 21, 34, 55]);
+  });
+});
 
 // b) Create the function that makes the test pass.
 
+// 1b) Pseudo code:
+// Declare a function named fibonator (const declaration)
+// Name the parameter as length
+// Declare a variable named fibArray, and initialize it as array [1, 1] (let declaration)
+// Create a for loop which iterates a number of times equal to the value of parameter length
+// for loop logic:
+//      Create a variable named fibValue which stores a sum equal the sum of the previous two elements in fibArray
+//      .push each value into fibArray
+// return fibArray
+// Run the test code to ensure it passes green
+// Think about refactoring
+
+const fibonator = (length) => {
+  let fibArray = [1, 1];
+  if (length < 2 || typeof length !== "number") {
+    return "Invalid input, please input a value greater than 1";
+  }
+  for (let i = 2; i < length; i++) {
+    let fibValue = fibArray[i - 2] + fibArray[i - 1];
+    fibArray.push(fibValue);
+  }
+  return fibArray;
+};
+
```

Comment:
Nice work! I agree with your drive to use map, but the thing about map is you need an array to iterate on. Recursion is def an option.